dags/synapse-dataset-to-croissant.py

```diff
@@ -165,17 +199,552 @@ def set_up_logging() -> Logger:
+
+    Returns:
+        None
+    """
+    try:
+        if not push_to_synapse:
+            otel_logger.info(
+                f"Croissant file link for [dataset: {dataset.name}, id: {dataset_id}.{dataset_version}]: {s3_url}")
+            return
+
+        otel_logger.info(
+            f"Uploading croissant file link to Synapse table {SYNAPSE_TABLE_FOR_CROISSANT_LINKS}"
+        )
+
+        # TODO: When 4.8.0 of the SYNPY client is released this may be
+        # replaced by the upsert functionality
+
+        # Warning: Using an authenticated Synapse Client during this section of code
+        syn_hook = SynapseHook(
+            context["params"]["synapse_conn_id"])
+        authenticated_syn_client: Synapse = syn_hook.client
+        authenticated_syn_client._rest_call = MethodType(
+            _rest_call_replacement, authenticated_syn_client)
+        existing_row = authenticated_syn_client.tableQuery(
+            query=f"SELECT * FROM {SYNAPSE_TABLE_FOR_CROISSANT_LINKS} WHERE dataset = '{dataset_id}' AND dataset_version = {dataset_version} AND dataset_collection = '{dataset_collection}'", resultsAs="csv")
+        existing_row_df = existing_row.asDataFrame()
+        os.remove(existing_row.filepath)
+        if not existing_row_df.empty and existing_row_df["croissant_file_s3_object"].values[0] == s3_url:
+            otel_logger.info(
+                f"Croissant file link already exists in Synapse table {SYNAPSE_TABLE_FOR_CROISSANT_LINKS}. Skipping.")
+            return
+
+        df = DataFrame(
+            data={
+                "dataset": [dataset_id],
+                "dataset_version": [dataset_version],
+                "dataset_collection": [dataset_collection],
+                "croissant_file_s3_object": [s3_url]
+            }
+        )
+        temp_dir = tempfile.mkdtemp()
+        filepath = os.path.join(temp_dir, "table.csv")
+
+        schema = authenticated_syn_client.get(
+            SYNAPSE_TABLE_FOR_CROISSANT_LINKS)
+        if existing_row_df.empty:
+            # If the row does not exist, create a new row
+            authenticated_syn_client.store(
+                Table(schema=schema, values=df, filepath=filepath))
+        else:
+            # Update the existing row with the new value
+            existing_row_df["croissant_file_s3_object"] = [s3_url]
+            authenticated_syn_client.store(
+                Table(schema=schema, values=existing_row_df, filepath=filepath))
+
+        os.remove(filepath)
+    except Exception as ex:
+        otel_logger.exception(
+            "Failed to push croissant file link to Synapse.")
+        otel_tracer.span_processor.force_flush()
+        otel_logger.handlers[0].flush()
+        raise ex
+
+
 @dag(**dag_config)
 def dataset_to_croissant() -> None:
-    """Execute a query on Snowflake and report the results to a Synapse table."""
+    """Execute a query on Snowflake and report the results to a Synapse table.
+
+
+    DAG Parameters:
```

Comment:
By Adding this to this docstring it will show up in the airflow UI now!

![image](https://github.com/user-attachments/assets/0dee4924-bb5c-42c8-b471-894955a19922)

Reply:
This, imo, is a good example of PR review comment that doesn't have to be in the code comment itself